pkg/rc/replication_controller.go

```diff
@@ -672,14 +672,14 @@ func (rc *replicationController) unschedule(txn *auditingTransaction, rcFields f
 // when a node transfer occurs
 func (rc *replicationController) attemptNodeTransfer(rcFields fields.RC, current types.PodLocations, ineligibles []types.NodeName, allocAttempts int) (bool, error) {
 	ok, unlocker, err := rc.canNodeTransfer(rcFields, current, ineligibles)
+	if unlocker != nil {
+		defer unlocker.Unlock()
+	}
```

Comment:
By moving this `defer` block, we increase the likelihood of unlocking the session if we hit one of our error cases.

Reply:
Are you sure the session will be destroyed when the consul session is unlocked? I think it needs to be explicitly destroyed. https://github.com/square/p2/blob/master/pkg/store/consul/session.go#L384

Reply:
Oh, I think you're right and I got stuck on trying to ensure it gets unlocked. Tackling that now.